### PR TITLE
update profile manual

### PR DESCRIPTION
**Pull request by KristofferC**
The profile section in the manual was terribly out of date. I tried to update it to the best of my abilities.

### CODE REVIEW

File: doc/src/manual/profile.md [hunk 1]
```diff
@@ -211,44 +225,31 @@ Let's discuss these arguments in order:
...
-    s = open("/tmp/prof.txt","w")
-    Profile.print(s,cols = 500)
-    close(s)
-    ```
-  * `maxdepth` can be used to limit the size of the output in `:tree` format (it nests only up to
-    level `maxdepth`)
-  * `sortedby = :count` sorts the `:flat` format in order of increasing counts
+```julia
+open("/tmp/prof.txt", "w") do s
+  Profile.print(IOContext(s, :displaysize => (24, 500)))
```
Comment (tkelman) [inline doc/src/manual/profile.md]: 4 space indent

File: doc/src/manual/profile.md [hunk 2]
```diff
@@ -194,14 +208,14 @@ be very useful, but sometimes you want to start fresh; you can do so with [`Prof
 [`Profile.print()`](@ref) has more options than we've described so far. Let's see the full declaration:
 
 ```julia
-function print(io::IO = STDOUT, data = fetch(); format = :tree, C = false, combine = true, cols = tty_cols(), maxdepth = typemax(Int), sortedby = :filefuncline)
+function print(io::IO = STDOUT, data = fetch(); kwargs...)
 ```
 
-Let's discuss these arguments in order:
+Let's first discuss the two first arguments and then later all the different possible keyword arguments:
```
Comment (KristofferC) [inline doc/src/manual/profile.md]: `two first` -> `first two`?

File: doc/src/manual/profile.md (hunk 1 above)
Comment (KristofferC) [inline doc/src/manual/profile.md]: fixed

File: doc/src/manual/profile.md [hunk 3]
```diff
@@ -61,80 +61,84 @@ available, but here we'll use the text-based display that comes with the standar
...
+        25 ./reduce.jl:428; mapreduce_impl(::Base.#identity, ::Base.#scalarmax, ::Array{F...
 ```
 
 Each line of this display represents a particular spot (line number) in the code. Indentation
 is used to indicate the nested sequence of function calls, with more-indented lines being deeper
 in the sequence of calls. In each line, the first "field" indicates the number of backtraces
 (samples) taken *at this line or in any functions executed by this line*. The second field is
-the file name, followed by a semicolon; the third is the function name followed by a semicolon,
-and the fourth is the line number. Note that the specific line numbers may change as Julia's
+the file name, followed by a colon and the line number; the third is the function name.
```
Comment (Sacha0) [inline doc/src/manual/profile.md]: Perhaps  "The second field provides the file name and line number. The third field [is|provides] the function name.", or instead "The second field provides the file name and line number, separate by a colon and terminated by a semicolon. The third field..." if you feel describing the field's punctuation is worthwhile?

File: doc/src/manual/profile.md [hunk 4]
```diff
@@ -61,80 +61,84 @@ available, but here we'll use the text-based display that comes with the standar
...
+Note that the specific line numbers may change as Julia's
 code changes; if you want to follow along, it's best to run this example yourself.
 
-In this example, we can see that the top level is `client.jl`'s `_start` function. This is the
-first Julia function that gets called when you launch Julia. If you examine line 373 of `client.jl`,
-you'll see that (at the time of this writing) it calls `run_repl()`, mentioned on the second line.
-This in turn calls `eval_user_input()`. These are the functions in `client.jl` that interpret
+In this example, we can see that the top level function called is in the file `event.jl`. This is the
+function that schedules the task that runs the REPL. If you examine line 97 of `REPL.jl`,
```
Comment (Sacha0) [inline doc/src/manual/profile.md]: The former version of this sentence ("This is the first Julia function that gets called when you launch Julia.") might be easier for a new reader to digest (than "schedules the task that runs the REPL")? Perhaps some simplification would be possible and worthwhile?

File: doc/src/manual/profile.md [hunk 5]
```diff
@@ -61,80 +61,84 @@ available, but here we'll use the text-based display that comes with the standar
...
+you'll see this calls `eval_user_input()`. This is the function that interprets
```
Comment (Sacha0) [inline doc/src/manual/profile.md]: Perhaps "you'll see this function calls" for clarity? Likewise perhaps "`eval_user_input()` interprets what you type..."?

File: doc/src/manual/profile.md [hunk 6]
```diff
@@ -61,80 +61,84 @@ available, but here we'll use the text-based display that comes with the standar
...
 
-`none` refers to the fact that we defined `myfunc` in the REPL, rather than putting it in a file;
-if we had used a file, this would show the file name. Line 2 of `myfunc()` contains the call to
-`rand`, and there were 8 (out of 23) backtraces that occurred at this line. Below that, you can
-see a call to `dsfmt_gv_fill_array_close_open!()` inside `dSFMT.jl`. You might be surprised not
-to see the `rand` function listed explicitly: that's because `rand` is *inlined*, and hence doesn't
-appear in the backtraces.
+`REPL` refers to the fact that we defined `myfunc` in the REPL, rather than putting it in a file;
+if we had used a file, this would show the file name. The `[1]` shows that the function `myfunc`
+was the first expression that got evaluated in the REPL. Line 2 of `myfunc()` contains the call to
```
Comment (Sacha0) [inline doc/src/manual/profile.md]: Perhaps "was the first expression evaluated in the REPL"?

File: doc/src/manual/profile.md (hunk 2 above)
Comment (Sacha0) [inline doc/src/manual/profile.md]: Perhaps "Let's first discuss the two positional arguments, and later the keyword arguments:"?

File: doc/src/manual/profile.md [hunk 7]
```diff
@@ -194,14 +208,14 @@ be very useful, but sometimes you want to start fresh; you can do so with [`Prof
...
 ```julia
-function print(io::IO = STDOUT, data = fetch(); format = :tree, C = false, combine = true, cols = tty_cols(), maxdepth = typemax(Int), sortedby = :filefuncline)
+function print(io::IO = STDOUT, data = fetch(); kwargs...)
 ```
 
-Let's discuss these arguments in order:
+Let's first discuss the two first arguments and then later all the different possible keyword arguments:
 
-  * The first argument allows you to save the results to a file, but the default is to print to `STDOUT`
+  * The first argument `io` allows you to save the results to a buffer, e.g. a file, but the default is to print to `STDOUT`
```
Comment (Sacha0) [inline doc/src/manual/profile.md]: Should `io` be offset by parens or commas, or are parens or commas unnecessary in this case? (I do not know the rules for this case.)

File: doc/src/manual/profile.md [hunk 8]
```diff
@@ -211,44 +225,31 @@ Let's discuss these arguments in order:
...
-         +4 5 ...ence.jl; ...ct_eval_call; line: 733
-         +6 5 ...ence.jl; abstract_call; line: 645
-    ```
+The keyword arguments can be any combination of:
 
-    File/function names are sometimes truncated (with `...`), and indentation is truncated with a
-    `+n` at the beginning, where `n` is the number of extra spaces that would have been inserted,
-    had there been room. If you want a complete profile of deeply-nested code, often a good idea is
-    to save to a file and use a very wide `cols` setting:
+  * `format` which was introduced above. The possible choices are `:tree` and
```
Comment (Sacha0) [inline doc/src/manual/profile.md]: Perhaps "`format`, which was introduced above." or simply "`format`, introduced above."?

File: doc/src/manual/profile.md [hunk 9]
```diff
@@ -211,44 +225,31 @@ Let's discuss these arguments in order:
...
+    `:flat`.
+  * If `C` is `true`, backtraces from C and Fortran code are shown (normally they are excluded).
```
Comment (Sacha0) [inline doc/src/manual/profile.md]: Why remove the additional explanation for keyword arg `C`? Seems helpful?

Might be nice to consistently begin each list item with the item's name?

File: doc/src/manual/profile.md [hunk 10]
```diff
@@ -211,44 +225,31 @@ Let's discuss these arguments in order:
...
+  * If `combine` is `true` (default), instruction pointers are merged that correspond to the same line of code.
```
Comment (Sacha0) [inline doc/src/manual/profile.md]: Likewise with `combine`, why remove the additional explanation? Seems helpful?

File: doc/src/manual/profile.md [hunk 11]
```diff
@@ -211,44 +225,31 @@ Let's discuss these arguments in order:
...
+  * `maxdepth` can be used to limit the depth of printing in `:tree` format
```
Comment (Sacha0) [inline doc/src/manual/profile.md]: Perhaps "`maxdepth` limits printing depth in `:tree` format"? More active?

File: doc/src/manual/profile.md [hunk 12]
```diff
@@ -211,44 +225,31 @@ Let's discuss these arguments in order:
...
+  * `sortedby` can be used to control the order in `:flat` format. `:filefuncline` (default) sorts by the source
```
Comment (Sacha0) [inline doc/src/manual/profile.md]: Likewise, perhaps "`sortedby` controls the ...", and perhaps "`mincount` limits the printout..." below?

**Review by Sacha0 — COMMENTED** [review]
This looks great! Please see the associated minor review comments. Best!

File: doc/src/manual/profile.md (hunk 10 above)
Comment (KristofferC) [inline doc/src/manual/profile.md]: I mostly replaced the docs for the keyword arguments with the one that are in the inline docstrings for `Profile.print`. I will try to extract the best from both and merge them better than now.

**Comment by KristofferC** [comment]
Thanks a lot for review @Sacha0. I made some updates (both from your comments and some on my own initiative). Happy to keep iterating.

File: doc/src/manual/profile.md [hunk 13]
```diff
@@ -61,80 +61,84 @@ available, but here we'll use the text-based display that comes with the standar
...
+function that runs the REPL when you launch Julia. If you examine line 97 of `REPL.jl`,
+you'll see this is where the function `eval_user_input()` is called. This is the function that evaluates
 what you type at the REPL, and since we're working interactively these functions were invoked
 when we entered `@profile myfunc()`. The next line reflects actions taken in the [`@profile`](@ref)
 macro.
 
-The first line shows that 23 backtraces were taken at line 373 of `client.jl`, but it's not that
-this line was "expensive" on its own: the second line reveals that all 23 of these backtraces
-were actually triggered inside its call to `run_repl`, and so on. To find out which operations
+The first line shows that 80 backtraces were taken at line 73 of `client.jl`, but it's not that
```
Comment (jebej) [inline doc/src/manual/profile.md]: of `event.jl`

File: doc/src/manual/profile.md [hunk 14]
```diff
@@ -211,44 +225,37 @@ Let's discuss these arguments in order:
...
-      33 inference.jl; abstract_call; line: 645
-        33 ...rence.jl; abstract_call_gf; line: 567
-           33 ...nce.jl; typeinf; line: 1201
-         +1 5  ...nce.jl; ...t_interpret; line: 900
-         +3 5 ...ence.jl; abstract_eval; line: 758
-         +4 5 ...ence.jl; ...ct_eval_call; line: 733
-         +6 5 ...ence.jl; abstract_call; line: 645
-    ```
+    on this line. `combine = true` lumps them together, and is probably what you typically want, but
+    you can generate an output separately for each unique instruction pointer with `combine = false`
```
Comment (Sacha0) [inline doc/src/manual/profile.md]: Missing a terminating period?

File: doc/src/manual/profile.md [hunk 15]
```diff
@@ -211,44 +225,37 @@ Let's discuss these arguments in order:
...
+  * `maxdepth` -- limits frames at a depth higher than `maxdepth` in the `:tree` format
```
Comment (Sacha0) [inline doc/src/manual/profile.md]: Capitalize "Limits" and terminate with a period for consistency?

**Review by Sacha0 — APPROVED** [review]
> Thanks a lot for review `@Sacha0`.

A pleasure; thanks for the PR! LGTM modulo minor consistency comments. Best!

File: doc/src/manual/profile.md (hunk 13 above)
Comment (KristofferC) [inline doc/src/manual/profile.md]: Thanks, fixed.

**Comment by KristofferC** [comment]
Updated from latest review. Thanks again.

File: base/profile.jl [hunk 16]
```diff
@@ -103,23 +103,22 @@ will be used.
 
 The keyword arguments can be any combination of:
 
- - `format` can be `:tree` (default) or `:flat`.
+ - `format` -- Determines how the backtraces are printed. Can be `:tree` (default) or `:flat`.
```
Comment (KristofferC) [inline base/profile.jl]: This is perhaps not so good since all the arguments here determine how backtraces are printed...

Comment (Sacha0) [inline base/profile.jl]: Perhaps "Determines whether backtraces are printed with (default, `:tree`) or without (`:flat`) indentation indicating tree structure."?

Comment (KristofferC) [inline base/profile.jl]: Beautiful

Comment (KristofferC) [inline base/profile.jl]: Updated.

File: base/profile.jl [hunk 17]
```diff
@@ -103,7 +103,8 @@ will be used.
 
 The keyword arguments can be any combination of:
 
- - `format` -- Determines how the backtraces are printed. Can be `:tree` (default) or `:flat`.
+ - `format` -- Determines whether backtraces are printed with (default, :tree) or without (:flat)
```
Comment (tkelman) [inline base/profile.jl]: code highlight the `:tree` and `:flat`

Comment (KristofferC) [inline base/profile.jl]: Oh yeah, done.

**Pull request merged**